iroha/src/peer.rs

```diff
@@ -10,6 +10,21 @@ pub mod isi {
     use super::*;
     use iroha_data_model::prelude::*;
 
+    impl Execute for Register<Peer, Peer> {
+        fn execute(
+            self,
+            _authority: <Account as Identifiable>::Id,
+            world_state_view: &WorldStateView,
+        ) -> Result<WorldStateView, String> {
+            let mut world_state_view = world_state_view.clone();
+            let _ = world_state_view
+                .peer()
+                .trusted_peers_ids
+                .push(self.object.id);
```

Comment:
I think we should check if the peer is in trusted peers and if it's already there, fail this ISI. Otherwise if we have peer duplicates consensus will fail.

Reply:
@eadventurous you are right, I will replace vector with set to prevent a possibility of duplicates.

Reply:
We could still fail the ISI even when we have Set there, just to give feedback to the user that the peer is already there.

Reply:
sure